apps/system/js/app_usage_metrics.js

```diff
@@ -613,19 +592,77 @@
     };
 
     var deviceInfoQuery = {
-      'deviceinfo.update_channel': 'unknown',
+      'app.update.channel': 'unknown',
+      'developer.menu.enabled': false, // If true, data is probably an outlier
+      'deviceinfo.hardware': 'unknown',
+      'deviceinfo.os': 'unknown',
       'deviceinfo.platform_version': 'unknown',
       'deviceinfo.platform_build_id': 'unknown',
-      'developer.menu.enabled': false // If true, data is probably an outlier
+      'deviceinfo.product_model': 'unknown',
+      'deviceinfo.software': 'unknown',
+      'deviceinfo.update_channel': 'unknown'
     };
 
     // Query the settings db to get some more device-specific information
-    AUM.getSettings(deviceInfoQuery, function(deviceinfo) {
-      data.deviceinfo = deviceinfo;
+    AUM.getSettings(deviceInfoQuery, function(deviceInfo) {
+      data.deviceinfo = deviceInfo;
+      data.simInfo = getSIMInfo();
+
       // Now transmit the data
       send(data);
     });
 
+    function getSIMInfo() {
+      var simInfo = {
+        network: null,
+        icc: null
+      };
+
+      if (SIMSlotManager.noSIMCardConnectedToNetwork()) {
+        // No connected SIMs
+        return simInfo;
+      }
+
+      var slots = SIMSlotManager.getSlots().filter(function(slot) {
+        return !slot.isAbsent() && !slot.isLocked();
+      });
+
+      if (slots.length === 0) {
+        // No unlocked or active SIM slots
+        return simInfo;
+      }
+
+      var conn = slots[0].conn;
+      if (!conn) {
+        // No connection
+        return simInfo;
+      }
+
+      var iccObj = navigator.mozIccManager.getIccById(conn.iccId);
+      var iccInfo = iccObj ? iccObj.iccInfo : null;
+      var voiceNetwork = conn.voice ? conn.voice.network : null;
+      if (!iccInfo && !voiceNetwork) {
+        // No voice network or ICC info
+        return simInfo;
+      }
+
+      simInfo.network = MobileOperator.userFacingInfo(conn);
+      if (voiceNetwork) {
+        simInfo.network.mnc = voiceNetwork.mnc;
+        simInfo.network.mcc = voiceNetwork.mcc;
+      }
+
+      if (iccInfo) {
+        simInfo.icc = {
+          mnc: iccInfo.mnc,
+          mcc: iccInfo.mcc,
+          spn: iccInfo.spn
+        };
```

Comment:
Are there privacy implications here with this data?  I don't know much about sims... Is mnc like the US area code?  

I gather that mcc and spn pretty much just identify the carrier, and that seems okay. I'm not sure about mnc, though.